src/App.tsx

```diff
@@ -1,61 +1,140 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+function getUserByName(userName: string): User | null {
+  return usersFromServer.find(user => user.name === userName) || null;
+}
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => (
+  {
+    ...todo,
+    user: getUserById(todo.userId),
+  }
+));
 
 export const App = () => {
+  const [todos, setTodos] = useState(preparedTodos);
+  const [title, setTitle] = useState('');
+  const [userName, setUserName] = useState('');
+  const [shouldTitleError, setShouldTitleError] = useState(false);
+  const [shouldUserNameError, setShouldUserNameError] = useState(false);
+
+  const handleTitile = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
noticed a minor typo
```suggestion
  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Reply:
Thank you 😅